### PR TITLE
feat(replication): allow non-tls connections between replica and master #1419 

### PR DESCRIPTION
**Pull request by kostasrim**
Addresses #1419 and form the basis for #1418.

1. Allow non-tls connections between replica and master on admin port
2. Add test
3. Add tls certificate generator

### CODE REVIEW

File: .github/workflows/regression-tests.yml [hunk 1]
```diff
@@ -63,7 +61,34 @@ jobs:
...
+          # Step 2
+          # Generate Dragonfly's private key and certificate signing request (CSR)
+          openssl req -newkey rsa:4096 -nodes -keyout df-key.pem -out df-req.pem -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"
+
+          # Step 3
+          # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+          openssl x509 -req -in df-req.pem -days 1 -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out df-cert.pem -extfile df-ext.cnf
 
+          # Start the test
+          pytest --json-report --json-report-file=rep1_report.json -sv dragonfly/tls_replication_test.py --df alsologtostderr
```
Comment (kostasrim) [inline .github/workflows/regression-tests.yml:91]: I should run these manually and verify.

File: tests/dragonfly/tls_replication_test.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    with_tls_args = tls_args(master)
+    # 2. Spin up dragonfly again, this time, with different options:
+    #   a. Enable tls
+    #   b. Allow non-tls connection from replica to master
+    master = df_local_factory.create(**with_tls_args, port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    # we need to sleep, because currently tls connections on master fail
+    # TODO Fix this once master properly loads tls certificates and redis-cli does not
+    # require --insecure flag, therefore for now we can't use `wait_available`
+    time.sleep(10)
```
Comment (kostasrim) [inline tests/dragonfly/tls_replication_test.py:50]: Not a big fun of this, it should take a couple of `ms` to load the snapshot so I don't expect the code below to fail. 

This will be fixed once: 
1. Server works properly with TLS && redis-cli connects without the --insecure flag

File: src/facade/dragonfly_connection.cc [hunk 3]
```diff
@@ -297,7 +299,15 @@ void Connection::UnregisterShutdownHook(ShutdownHandle id) {
...
 bool Connection::IsReplicaCommand(util::tls::TlsSocket::Buffer buff) const {
   string_view str{reinterpret_cast<const char*>(buff.data()), buff.size()};
-  return str.starts_with("PING") || str.starts_with("DFLY") || str.starts_with("REPLCONF");
+  auto starts_with = [](const string_view source, const string_view starter) {
+    if (source.size() < starter.size()) {
+      return false;
+    }
+    size_t pos = 0;
+    return std::all_of(source.begin(), source.begin() + starter.size(),
+                       [&pos, starter](const char c) mutable { return c == starter[pos++]; });
```
Comment (chakaz) [inline src/facade/dragonfly_connection.cc:308]: I'd replace these lines with:
```cpp
return source.substr(0, starter.size()) == starter);
```

Comment (chakaz) [reply src/facade/dragonfly_connection.cc:308]: Or better yet: use `absl::StartsWith()` (https://github.com/abseil/abseil-cpp/blob/master/absl/strings/match.h#L58)

File: .github/workflows/regression-tests.yml (hunk 1 above)
Comment (royjacobson) [reply .github/workflows/regression-tests.yml:91]: Can you hardcode the certificate into `tls_replication_test.py` and write the pem/crt into a temporary dir from there? I don't like this special pipeline and it makes running it locally harder.

File: src/facade/dragonfly_connection.cc [hunk 4]
```diff
@@ -36,6 +41,10 @@ ABSL_FLAG(std::string, admin_bind, "",
 ABSL_FLAG(std::uint64_t, request_cache_limit, 1ULL << 26,  // 64MB
           "Amount of memory to use for request cache in bytes - per IO thread.");
 
+ABSL_FLAG(bool, no_tls_on_replica, false,
+          "Enable one way TLS for master only, that is, replica does not use TLS to"
```
Comment (royjacobson) [inline src/facade/dragonfly_connection.cc:45]: What do you mean by 'one way TLS'?

**Comment by royjacobson** [comment]
General comment about this: Whether we want TLS is a question of what network interface we listen on and its security. If the network doesn't need TLS then we don't need TLS at all, if the network needs TLS then we should always have TLS. From a security perspective I really don't like that it is specific to replication.

File: src/facade/dragonfly_connection.cc (hunk 3 above)
Comment (kostasrim) [reply src/facade/dragonfly_connection.cc:308]: I was expecting this comment here since you mentioned absl with `StartsWith`  on my other PR

File: src/facade/dragonfly_connection.cc (hunk 4 above)
Comment (kostasrim) [reply src/facade/dragonfly_connection.cc:45]: Meaning only the master uses TLS while the client (the replica) does not. Maybe it's a confusing term and I should better change it

File: .github/workflows/regression-tests.yml (hunk 1 above)
Comment (kostasrim) [reply .github/workflows/regression-tests.yml:91]: That's a really good idea. +1

**Comment by kostasrim** [comment]
@royjacobson 

>General comment about this: Whether we want TLS is a question of what network interface we listen on and its security. If the network doesn't need TLS then we don't need TLS at all, if the network needs TLS then we should always have TLS. From a security perspective I really don't like that it is specific to replication.

Why do you think it's a security issue if it's specific for replication? If we are on a private network then why would the replica need to communicate over tls with master? (I think this was the original idea). 

I am not an expert in security, so plz add your concerns here and we can chat a little bit tomorrow on tomorrow's standup.

**Comment by royjacobson** [comment]
> @royjacobson
> 
> > General comment about this: Whether we want TLS is a question of what network interface we listen on and its security. If the network doesn't need TLS then we don't need TLS at all, if the network needs TLS then we should always have TLS. From a security perspective I really don't like that it is specific to replication.
> 
> Why do you think it's a security issue if it's specific for replication? If we are on a private network then why would the replica need to communicate over tls with master? (I think this was the original idea).
> 
> I am not an expert in security, so plz add your concerns here and we can chat a little bit tomorrow on tomorrow's standup.

My problem is that this is an unclean security abstraction: If the network is private then all connections should be without TLS. Making this relaxation specific for replication creates a difference in the network security level that is based on application logic and not on the network logic.

File: src/facade/dragonfly_connection.cc [hunk 5]
```diff
@@ -288,6 +297,19 @@ void Connection::UnregisterShutdownHook(ShutdownHandle id) {
   }
 }
 
+bool Connection::IsReplicaCommand(util::tls::TlsSocket::Buffer buff) const {
```
Comment (romange) [inline src/facade/dragonfly_connection.cc:300]: So this PR is not relevant anymore, right?

Comment (kostasrim) [reply src/facade/dragonfly_connection.cc:300]: some of it. The tests and notls flag on replica is useful -- I will push the clean up today

File: src/facade/dragonfly_connection.cc [hunk 6]
```diff
@@ -300,22 +309,23 @@ void Connection::HandleRequests() {
...
+
+      if (!aresult) {
+        LOG(WARNING) << "Error handshaking " << aresult.error().message();
+        return;
+      }
+      VLOG(1) << "TLS handshake succeeded";
     }
-    VLOG(1) << "TLS handshake succeeded";
   }
-  FiberSocketBase* peer = tls_sock ? (FiberSocketBase*)tls_sock.get() : socket_.get();
```
Comment (adiholden) [inline src/facade/dragonfly_connection.cc:316]: why was this removed?

File: src/facade/dragonfly_connection.h [hunk 7]
```diff
@@ -14,11 +14,13 @@
 #include "base/io_buf.h"
 #include "util/connection.h"
 #include "util/http/http_handler.h"
+#include "util/tls/tls_engine.h"
```
Comment (adiholden) [inline src/facade/dragonfly_connection.h:17]: why are this changes in header needed now?

File: src/facade/dragonfly_connection.cc (hunk 6 above)
Comment (kostasrim) [reply src/facade/dragonfly_connection.cc:316]: Because I simplified it a little bit, it gets initialized at the top :)

File: tests/dragonfly/tls_replication_test.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    #   a. Enable tls
+    #   b. Allow non-tls connection from replica to master
+    master = df_local_factory.create(admin_port=ADMIN_PORT, admin_nopass=True, **with_tls_args, port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    # we need to sleep, because currently tls connections on master fail
+    # TODO Fix this once master properly loads tls certificates and redis-cli does not
+    # require --insecure flag, therefore for now we can't use `wait_available`
+    time.sleep(10)
+
+    # 3. Try to connect with redis cli on master. This should fail since redis-cli does
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:83]: this is not redis cli :)

File: tests/dragonfly/tls_replication_test.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    # not use tls key and certificate
+    c_master = aioredis.Redis(port=master.port)
+    try:
+        await c_master.execute_command("DBSIZE")
+        raise "Non tls connection connected on master with tls. This should NOT happen"
+    except redis.ConnectionError:
+        pass
+
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:91]: lets add a connection via admin port and run DBSIZE command, see that it passes

Comment (adiholden) [reply tests/dragonfly/tls_replication_test.py:91]: this can also be a test in connection_test.py as checking that running command via admin port does not require tls is not related to replication

File: src/facade/dragonfly_connection.h (hunk 7 above)
Comment (kostasrim) [reply src/facade/dragonfly_connection.h:17]: The one below is needed, it's used in the interface and it was not included. It's a good practice to include headers that are part of your interface and avoid implicit includes

File: tests/dragonfly/tls_replication_test.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    # 4. Spin up a replica and initiate a REPLICAOF
+    replica = df_local_factory.create(port=BASE_PORT + 1, proactor_threads=t_replica,)
+    replica.start()
+    c_replica = aioredis.Redis(port=replica.port)
+    res = await c_replica.execute_command("REPLICAOF localhost " + str(ADMIN_PORT))
+    assert b"OK" == res
+    time.sleep(10)
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:98]: instead of sleep
use
await check_all_replicas_finished([c_replica], c_master)
The clients here should be connected to admin port

File: tests/dragonfly/tls_replication_test.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    tls_server_cert = df_instance.dfly_path + tls_server_cert_file_name;
+    step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
+    subprocess.run(step3, shell=True)
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_master, t_replica", replication_cases)
+@dfly_args({"dbfilename": "test-tls-replication-{{timestamp}}"})
+async def test_replication_all(df_local_factory, df_seeder_factory, t_master, t_replica):
+    # 1. Spin up dragonfly without tls, debug populate and then shut it down
+    master = df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:62]: I think that this first step is redundant, as you can run the dubug populate command from admin port when starting master with the tls args

File: tests/dragonfly/tls_replication_test.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+    master.stop()
+    with_tls_args = tls_args(master)
+    gen_tls_cert(master)
+    # 2. Spin up dragonfly again, this time, with different options:
+    #   a. Enable tls
+    #   b. Allow non-tls connection from replica to master
+    master = df_local_factory.create(admin_port=ADMIN_PORT, admin_nopass=True, **with_tls_args, port=BASE_PORT, proactor_threads=t_master)
+    master.start()
+    # we need to sleep, because currently tls connections on master fail
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:78]: we need to sleep because lts connection fail? I dont understand this.. where do you create connection? how is sleeping helps here?

File: tests/dragonfly/tls_replication_test.py [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+        await c_master.execute_command("DBSIZE")
+        raise "Non tls connection connected on master with tls. This should NOT happen"
+    except redis.ConnectionError:
+        pass
+
+    # 4. Spin up a replica and initiate a REPLICAOF
+    replica = df_local_factory.create(port=BASE_PORT + 1, proactor_threads=t_replica,)
+    replica.start()
+    c_replica = aioredis.Redis(port=replica.port)
+    res = await c_replica.execute_command("REPLICAOF localhost " + str(ADMIN_PORT))
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:96]: "REPLICAOF localhost {master.admin_port}"

File: tests/dragonfly/tls_replication_test.py (hunk 9 above)
Comment (kostasrim) [reply tests/dragonfly/tls_replication_test.py:91]: I ended up having it here. It's not related to replication but it doesn't require to create a new test case for that. It's small detail ;)

File: src/facade/dragonfly_connection.cc (hunk 6 above)
Comment (romange) [reply src/facade/dragonfly_connection.cc:316]: how is it working?
I would expect the test you wrote to fail.

File: tests/dragonfly/tls_replication_test.py (hunk 9 above)
Comment (romange) [reply tests/dragonfly/tls_replication_test.py:79]: hmm, I kind of agree with Adi here. The feature here is being able to connect to the admin port without using the tls. I think the slave part is unnecessary machinery for this test.

File: tests/dragonfly/tls_replication_test.py [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+tls_server_cert_file_name = "df-cert.pem"
+
+def tls_args(dfly_path):
+    with_tls_args = {"tls": "",
+                    "tls_key_file": dfly_path + tls_server_key_file_name,
+                    "tls_cert_file": dfly_path + tls_server_cert_file_name,
+                    "no_tls_on_admin_port": "true"}
+    return with_tls_args
+
+def gen_tls_cert(dfly_path):
```
Comment (romange) [inline tests/dragonfly/tls_replication_test.py:32]: consider making it a fixture (in conftest.py) similarly to other resources being consumed by tests.

File: src/facade/dragonfly_connection.h [hunk 15]
```diff
@@ -19,6 +19,7 @@
 #include "core/fibers.h"
 #include "facade/facade_types.h"
 #include "facade/resp_expr.h"
+#include "util/tls/tls_socket.h"
```
Comment (romange) [inline src/facade/dragonfly_connection.h:22]: why did you need it here?

Comment (kostasrim) [reply src/facade/dragonfly_connection.h:22]: Oh it's `FiberSocketBase *`, I completely missed that, I thought it's used explicitly. removed.

File: src/facade/dragonfly_connection.cc (hunk 6 above)
Comment (kostasrim) [reply src/facade/dragonfly_connection.cc:316]: Indeed I messed this up on refactoring and HUGE thumps up for catching this. It works, because we only test the admin port and not the TLS connection itself -- we have 0 tests for that (although my other PR addresses this and it would have been caught there).

File: tests/dragonfly/tls_replication_test.py (hunk 9 above)
Comment (kostasrim) [reply tests/dragonfly/tls_replication_test.py:79]: Maybe I missed something but @adiholden asked to promt admin port directly (on previous commits that was not the case because of how I initially implemented this) -- which I addressed.

Now for (`# 2. Try to connect on master without admin port. This should fail.`), I think it's good to be here because it asserts that the changes/flags introduced that allowed non-tls connections on admin port did not also allow non-tls connections on other ports (it could be the case that I messed up the if conditions here: https://github.com/dragonflydb/dragonfly/pull/1490/files#diff-b88cbca1b2c9337ab2a37262f9d516b117f4996da25e2ae173e0e129f2b82b20R310) -- this guards as against that potential mistake and that's why I put it here.

File: tests/dragonfly/tls_replication_test.py (hunk 10 above)
Comment (adiholden) [reply tests/dragonfly/tls_replication_test.py:86]: please remove the sleep

File: tests/dragonfly/tls_replication_test.py [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    res = await c_replica.execute_command("REPLICAOF localhost " + str(master.admin_port))
+    assert b"OK" == res
+    time.sleep(10)
+    await check_all_replicas_finished([c_replica], c_master)
+
+    # 4. Verify that replica dbsize == debug populate key size -- replication works
+    db_size = await c_replica.execute_command("DBSIZE")
+    assert 100 == db_size
+
+    master.stop()
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:93]: I believe you dont need to call stop at the end of the test as this is done in df_local_factory

File: tests/dragonfly/tls_replication_test.py (hunk 9 above)
Comment (adiholden) [reply tests/dragonfly/tls_replication_test.py:79]: The comment here is that the changes in dfly code are not specific for replication, we enable connecting from admin port without tls. Therefor we would like to test this specifically and the replication test is another test to check that replication work with this, we can add this test but f.e if replication with no tls breaks this does not mean that connecting via admin port with no tls fails.
f.e if the PR was about adding admin port to dfly I would test it in connection test, it is importatnt to validate that replication as well works with admin port there fore I will add another test for replication with admin port
I believe that we need

File: tests/dragonfly/tls_replication_test.py (hunk 10 above)
Comment (kostasrim) [reply tests/dragonfly/tls_replication_test.py:86]: I would have taken an oath that I removed it, I really don't know how it got left -- removed

File: tests/dragonfly/connection_test.py [hunk 17]
```diff
@@ -2,6 +2,7 @@
 import pytest
 import asyncio
 from redis import asyncio as aioredis
+from redis.exceptions import ConnectionError
```
Comment (royjacobson) [inline tests/dragonfly/connection_test.py:5]: `ConnectionError` is a Python builtin so it's probably best not to override it

File: tests/dragonfly/connection_test.py [hunk 18]
```diff
@@ -415,3 +416,23 @@ async def test_large_cmd(async_client: aioredis.Redis):
 
     res = await async_client.mget([f"key{i}" for i in range(MAX_ARR_SIZE)])
     assert len(res) == MAX_ARR_SIZE
+
+
+@pytest.mark.asyncio
+@dfly_args({"admin_nopass" : True})
+async def test_reject_non_tls_connections_on_tls_master(gen_tls_cert, tls_server_key_file_name, tls_server_cert_file_name, df_local_factory, df_seeder_factory):
+    gen_tls_cert
```
Comment (royjacobson) [inline tests/dragonfly/connection_test.py:424]:
```suggestion
```
?

File: src/facade/dragonfly_connection.cc [hunk 19]
```diff
@@ -300,22 +302,24 @@ void Connection::HandleRequests() {
...
   if (ctx_) {
-    tls_sock.reset(new tls::TlsSocket(socket_.get()));
-    tls_sock->InitSSL(ctx_);
-
-    FiberSocketBase::AcceptResult aresult = tls_sock->Accept();
-    if (!aresult) {
-      LOG(WARNING) << "Error handshaking " << aresult.error().message();
-      return;
+    const bool no_tls_on_admin_port = absl::GetFlag(FLAGS_no_tls_on_admin_port);
+    if (!no_tls_on_admin_port || (no_tls_on_admin_port && !IsAdmin())) {
```
Comment (royjacobson) [inline src/facade/dragonfly_connection.cc:310]: I find this a bit more readable -
```suggestion
    if (!(IsAdmin() && no_tls_on_admin_port)) {
```

File: tests/dragonfly/__init__.py [hunk 20]
```diff
@@ -159,6 +159,13 @@ def stop_all(self):
     def __str__(self):
         return f"Factory({self.args})"
 
+    @property
+    def dfly_path(self):
```
Comment (royjacobson) [inline tests/dragonfly/__init__.py:163]: Maybe use `os.path.basedir` here? hard-coding the test binary name is a bit annoying if people try to bisect regressions etc

File: tests/dragonfly/connection_test.py [hunk 21]
```diff
@@ -415,3 +416,23 @@ async def test_large_cmd(async_client: aioredis.Redis):
...
+    master = df_local_factory.create(admin_port=1111, port=1211, **with_tls_args)
+    master.start()
+
+    # Try to connect on master without admin port. This should fail.
+    c_master = aioredis.Redis(port=master.port)
+    try:
+        await c_master.execute_command("DBSIZE")
+        raise "Non tls connection connected on master with tls. This should NOT happen"
+    except redis.exceptions.ConnectionError:
+         pass
```
Comment (adiholden) [inline tests/dragonfly/connection_test.py:438]: please connect via admin port also and check it success

File: tests/dragonfly/tls_replication_test.py [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+
+# 1. Number of master threads
+# 2. Number of threads for each replica
+replication_cases = [(8, 8)]
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_master, t_replica", replication_cases)
+@dfly_args({"dbfilename": "test-tls-replication-{{timestamp}}", "admin_nopass" : True})
+async def test_no_tls_on_admin_port(gen_tls_cert, tls_server_key_file_name, tls_server_cert_file_name, df_local_factory, df_seeder_factory, t_master, t_replica):
+    gen_tls_cert
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:25]: remove gen_tls_cert

File: tests/dragonfly/tls_replication_test.py [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+Test replication with tls and non-tls options
+"""
+
+# 1. Number of master threads
+# 2. Number of threads for each replica
+replication_cases = [(8, 8)]
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("t_master, t_replica", replication_cases)
+@dfly_args({"dbfilename": "test-tls-replication-{{timestamp}}", "admin_nopass" : True})
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:23]: I believe you can drop the dbfilename param
why is the admin_nopass needed?

File: tests/dragonfly/tls_replication_test.py [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+from . import DflyInstanceFactory, dfly_args
+import logging
+import time
+from dragonfly.replication_test import check_all_replicas_finished
+
+BASE_PORT = 1111
+ADMIN_PORT = 1211
+
+"""
+Test replication with tls and non-tls options
```
Comment (adiholden) [inline tests/dragonfly/tls_replication_test.py:14]: Actually this is just another replication test.. you can just move it to replicaiton_test.py

File: tests/dragonfly/conftest.py [hunk 25]
```diff
@@ -225,3 +226,40 @@ def port_picker():
...
+
+@pytest.fixture(scope="function")
+def tls_server_key_file_name():
+    return "df-key.pem"
+
+@pytest.fixture(scope="function")
+def tls_server_cert_file_name():
+    return "df-cert.pem"
+
+@pytest.fixture(scope="function")
```
Comment (adiholden) [inline tests/dragonfly/conftest.py:239]: I think its better to use scope="session" so this function will run only once when we are running regression tests

File: tests/dragonfly/tls_replication_test.py (hunk 23 above)
Comment (kostasrim) [reply tests/dragonfly/tls_replication_test.py:23]: to make our life easier so I don't have to setup a password

File: tests/dragonfly/conftest.py [hunk 26]
```diff
@@ -225,3 +226,40 @@ def port_picker():
...
+    tls_server_key = dfly_path + tls_server_key_file_name
+    tls_server_req = dfly_path + "df-req.pem"
+    step2 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
+    subprocess.run(step2, shell=True)
+
+    # Step 3
+    # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+    tls_server_cert = dfly_path + tls_server_cert_file_name
+    step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
+    subprocess.run(step3, shell=True)
```
Comment (adiholden) [inline tests/dragonfly/conftest.py:265]: gen_tls_cert can return the path of key and certificate file names and instead of defining tls_server_cert_file_name and tls_server_key_file_name set the filenames to constant var

File: tests/dragonfly/tls_replication_test.py (hunk 23 above)
Comment (adiholden) [reply tests/dragonfly/tls_replication_test.py:23]: why password is required if requirepass flag is not set?

Comment (kostasrim) [reply tests/dragonfly/tls_replication_test.py]: because from my understanding you still need to provide an auth token according to https://github.com/dragonflydb/dragonfly/issues/1181

Comment (kostasrim) [reply tests/dragonfly/tls_replication_test.py]: I quote:

>However for auth protected setups we still need to provide an auth token for clients that connect via the admin port.

**Review by adiholden — APPROVED** [review]
